### DISCUSSION TITLE
Request for Concrete Damage Plasticity Model in TFEL/MFRONT

### DISCUSSION THREAD

**Discussion by JAWilliams123** [thelfer/tfel#139] [General]
Hello Thomas,

I have searched online but have not been able to find a TFEL/MFRONT implementation of the concrete damage plasticity model based on the work done by Peter Grassl ([https://petergrassl.com/Research/DamagePlasticity/index.html](url)).
Is it possible to request that this material model be added to TFEL/MFRONT so that it can be accessed/used in CalculiX?

Many thanks,
JW

**Comment by thelfer** [comment]
@JAWilliams123 The first step is to write the equations of the model properly. Then we could discuss if it is feasible.

I could *help* for the implementation, but your contribution will be required. Also testing will be mainly on your side.

I generally recommend to consider a new entry in [the gallery](https://thelfer.github.io/tfel/web/gallery.html). See attached a template. You need to have pandoc installed:

```bash
$ pandoc -f markdown+tex_math_single_backslash --citeproc --toc ConcreteDamagePlasticity.md -o ConcreteDamagePlasticity.pdf
```

[ConcreteDamagePlasticity.zip](https://github.com/thelfer/tfel/files/8215363/ConcreteDamagePlasticity.zip)

**Reply by JAWilliams123** [reply]
Many thanks @thelfer,

Would the linked manuscript below be useful in determining the feasibility of the implementation? 
Link to CDPM2 [https://arxiv.org/pdf/1307.6998.pdf](url)

Best wishes,
JW

**Reply by thelfer** [reply]
@JAWilliams123, I had a very quick look at it.

The damage part seems more or less close to the Fichant-Larborderie behaviour:: <https://thelfer.github.io/tfel/web/FichantLaBorderieDamageBehaviour.html>.

The plastic part seems more complicated as it implies working with all the invariants. The Mohr-Coulumb example seems to provide the necessary ingredients: <https://thelfer.github.io/tfel/web/MohrCoulomb.html>

The implementation can be quite heavy, but is feasible. Testing will take at lot of time.

**Reply by JAWilliams123** [reply]
Many thanks @thelfer,

The information shared provides a good starting point for building the CDPM behaviour.

Best wishes,
JW

**Reply by thelfer** [reply]
If you are ok to participate to the MFront tutorials, then we could do that incrementally and see what happens.

**Reply by JAWilliams123** [reply]
Yes, I would not mind working on the implementation incrementally.

Best wishes,
JW

**Reply by thelfer** [reply]
No guarantee on my side. If you're ok with this and with publishing this work in the MFront gallery, let'us start with the damage part. Would you try to write the constitutive equations following the example of the Fichant-Larborderie behaviour ?
